diff --git a/node_modules/expo-modules-core/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/gradle/ExpoGradleHelperExtension.kt b/node_modules/expo-modules-core/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/gradle/ExpoGradleHelperExtension.kt
index 7c387ab..fb5d309 100644
--- a/node_modules/expo-modules-core/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/gradle/ExpoGradleHelperExtension.kt
+++ b/node_modules/expo-modules-core/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/gradle/ExpoGradleHelperExtension.kt
@@ -40,10 +40,13 @@ open class ExpoGradleHelperExtension {
       ?.projectDir
       ?.parentFile
 
+    val nodeExecutable = System.getenv("NODE_BINARY")
+      ?: System.getProperty("NODE_BINARY")
+      ?: "node"
     reactNativeDir = reactNativeDirFromSource ?: File(
       project.providers.exec { env ->
         env.workingDir(project.rootDir)
-        env.commandLine("node", "--print", "require.resolve('react-native/package.json')")
+        env.commandLine(nodeExecutable, "--print", "require.resolve('react-native/package.json')")
       }.standardOutput.asText.get().trim()
     ).parentFile
 